elements/lisk-cryptography/test/ed.spec.ts

```diff
@@ -30,219 +28,165 @@ import { MAX_UINT32 } from '../src/constants';
-					publicKey: Buffer.from(defaultPublicKey, 'hex'),
-				}),
+			expect(() =>
+				verifyMessageWithPublicKey({ message, signature: changeLength(signature), publicKey }),
 			).toThrow('Invalid signature length, expected 64-byte signature');
 		});
 
 		it('should return false if the signature is invalid', () => {
 			const verification = verifyMessageWithPublicKey({
-				message: defaultMessage,
-				signature: makeInvalid(Buffer.from(defaultSignature, 'hex')),
-				publicKey: Buffer.from(defaultPublicKey, 'hex'),
+				message,
+				signature: makeInvalid(signature),
+				publicKey,
 			});
-			expect(verification).toBe(false);
+			expect(verification).toBeFalse();
 		});
 
 		it('should return true if the signature is valid', () => {
-			const verification = verifyMessageWithPublicKey(defaultSignedMessage);
-			expect(verification).toBe(true);
+			const signedMessage = { message, publicKey, signature };
+			expect(verifyMessageWithPublicKey(signedMessage)).toBeTrue();
 		});
 	});
 
 	describe('#printSignedMessage', () => {
 		it('should wrap a single signed message into a printed Lisk template', () => {
-			const printedMessage = printSignedMessage({
-				message: defaultMessage,
-				signature: Buffer.from(defaultSignature, 'hex'),
-				publicKey: Buffer.from(defaultPublicKey, 'hex'),
-			});
-			expect(printedMessage).toBe(defaultPrintedMessage);
+			const generatedPrintedMessage = printSignedMessage({ message, signature, publicKey });
+			expect(generatedPrintedMessage).toBe(printedMessage);
 		});
 	});
 
 	describe('#signAndPrintMessage', () => {
 		it('should sign the message once and wrap it into a printed Lisk template', () => {
-			const signedAndPrintedMessage = signAndPrintMessage(
-				defaultMessage,
-				Buffer.from(defaultPrivateKey, 'hex'),
-			);
-			expect(signedAndPrintedMessage).toBe(defaultPrintedMessage);
+			const signedAndPrintedMessage = signAndPrintMessage(message, privateKey);
+			expect(signedAndPrintedMessage).toBe(printedMessage);
 		});
 	});
+});
 
-	describe('#signData', () => {
-		let signature: Buffer;
-
-		beforeEach(async () => {
-			signature = signData(tag, chainID, defaultData, Buffer.from(defaultPrivateKey, 'hex'));
-			return Promise.resolve();
-		});
+describe('sign and verify data', () => {
+	const tag = createMessageTag('TST');
+	const chainID = Buffer.from('10000000', 'hex');
 
-		it('should sign a transaction', () => {
-			expect(signature).toEqual(Buffer.from(defaultDataSignature, 'hex'));
-		});
-	});
+	const data = Buffer.from('This is some data');
+	const signature = signDataWithPrivateKey(tag, chainID, data, privateKey);
 
 	describe('#signDataWithPrivateKey', () => {
-		let signature: Buffer;
-
-		beforeEach(async () => {
-			signature = signDataWithPrivateKey(
-				tag,
-				chainID,
-				defaultData,
-				Buffer.from(defaultPrivateKey, 'hex'),
-			);
-			return Promise.resolve();
-		});
-
 		it('should sign a transaction', () => {
```

Comment:
I know it was not introduced in this PR, but isn't "a transaction" wrong as it is rather an arbitrary message?

Reply:
> Maybe that's what Shu meant in his previous message: Isn't tag a redundant argument since it will always be equal to MESSAGE_TAG_NON_PROTOCOL_MESSAGE?

It will not always have that value. It is a default value for the tag, if the value is not provided when invoking the function. User can provide a custom value for the tag, in order to verify any message.

My original implementation did not allow users to specify the tag. Then @shuse2 [requested to introduce that functionality](https://github.com/LiskHQ/lisk-sdk/pull/8805#discussion_r1284002053).